services/movie_session.py

```diff
@@ -0,0 +1,55 @@
+import datetime
+from db.models import MovieSession, Movie, CinemaHall
+from django.db.models.query import QuerySet
+
+
+def create_movie_session(
+        movie_show_time: datetime.datetime,
+        movie_id: int,
+        cinema_hall_id: int
+) -> QuerySet[MovieSession]:
+    return MovieSession.objects.create(
+        show_time=movie_show_time,
+        cinema_hall_id=cinema_hall_id,
+        movie_id=movie_id
+    )
+
+
+def get_movies_sessions(
+        session_date: datetime.date = None
+) -> QuerySet[MovieSession]:
+    queryset = MovieSession.objects.all()
+
+    if session_date:
+        queryset = queryset.filter(show_time__date=session_date)
+
+    return queryset
+
+
+def get_movie_session_by_id(movie_session_id: int) -> QuerySet[MovieSession]:
+    return MovieSession.objects.get(id=movie_session_id)
+
+
+def update_movie_session(
+        session_id: int,
+        show_time: str | None = None,
+        movie_id: int | None = None,
+        cinema_hall_id: int | None = None
+) -> None:
+    movie_session_to_update = MovieSession.objects.get(id=session_id)
+    if show_time:
+        movie_session_to_update.show_time = show_time
+
+    if movie_id is not None:
+        movie_session_to_update.movie = Movie.objects.get(id=movie_id)
```

Comment:
You can use: movie_session_to_update.movie_id = movie_id
No need to do additional queries

Reply:
Check in all places